### DISCUSSION TITLE
Best settings for PNG image compression

### DISCUSSION THREAD

**Discussion by stockydale** [ImageMagick/ImageMagick#4436] [Help]
Hello, I'm new here, so apologies for the stupid questions or if this has been asked before...

We are trying to compress some PNG images, aiming to optimize the best visual quality with the smallest file size.

Our original image is the 512x512 image that has been exported from a PDF - it is around 11KB
![13 - 512x512](https://user-images.githubusercontent.com/93651602/140047874-b108eec4-29dc-4a06-86a3-760bf526d00a.png)

We would like to reduce the image size by 50% to 256x256. So far we can get the image below, which is around 22KB
![13 - unsharp_1 4x1 4+5+0](https://user-images.githubusercontent.com/93651602/140048114-dc83bc62-1ea3-4bf4-93a7-f36fe039c571.png)

Here is an example of the same image that was compressed using another tool and it is only around 10KB and looks far better
![13](https://user-images.githubusercontent.com/93651602/140048286-a7d1e9e0-e64e-47a0-a806-b54afd402ab2.png)

So... my question is, are there any suggested settings that we should use to give us the best quality/size to try and achieve the output in the last image above?

Thanks!

**Comment by snibgo** [comment]
What version of IM, on what platform?

> We would like to reduce the image size by 50% to 256x256. So far we can get the image below, which is around 22KB.

What command did you use? Please always show your commands. I suspect you didn't do the obvious, which would be:
```
magick in.png -resize 50% out.png
```
Perhaps you used a low-quality method such as "-scale".

PNG is lossless, so after reducing the number of pixels, the compression method should make no difference to quality.

Five years ago I wrote up some results from the 100 different PNG quality settings: [Outputs: speed and size](http://im.snibgo.com/spdsiz.htm)

However, for smallest file size, I suggest using a program such as pngcrush.

**Reply by stockydale** [reply]
Hello, thanks for the quick reply... 

Here are some answers to your questions:
Version: ImageMagick 6.9.10-23 Q16 x86_64 20190101
System: Ubuntu 20.04.1
Command example: convert input.png `-resize 50% -unsharp 0x0.75+0.75+0.008 output.png` -> I tried with different values of unsharp parameter

`"0.25x0.25+8+0.065",
"0.5x0.5+5+0",
"1.0x1.0+0.5+0.1",
"1.0x1.0+5+0",
"1.2x1.2+5+0",
"1.2x1.2+8+0.1",
"1.5x1+0.7+0.02",
"1.4x1.4+5+0",
"1.6x1.6+3+0.1"`

I also ran the above files through pngcrush, but didn't only went from 22KB to 20KB.

It would be great to know if there are any other parameters we should experiment with.

Thanks

**Comment by snibgo** [comment]
To get a visually similar result to the "other tool", use "-scale 50%", with no unsharpening. The file is 11353 bytes. After a simple "pngcrush x.png", the size is 9405 bytes, which is slightly smaller then your "other tool".
